src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -263,7 +263,9 @@ internal class CSharpRecommendationService : AbstractRecommendationService
                 : context.SemanticModel.LookupSymbols(context.LeftToken.SpanStart);
 
             // Filter out any extension methods that might be imported by a using static directive.
-            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod());
+            // But include extension methods declared in the context's type
+            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod()
+            || symbol.ContainingType.DeclaringSyntaxReferences.Any(x => x.GetSyntax().IsEquivalentTo(context.ContainingTypeDeclaration))).ToList();
```

Comment:
Since you're allowing extensions from any partial part of the enclosing type (which seems right to me), you can just compare the `ContainingType` of each `symbol` with the result of `context.SemanticModel.GetEnclosingNamedType`.